src/System.Slices/System/Buffers/Memory.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Represents a Span<byte> factory</byte>
+    /// </summary>
+    /// <remarks>
+    /// This struct is not safe for multithreaded access, i.e. it's subject to struct tearing that can result in unsafe memory access. 
```

Comment:
Why? If it's immutable, we can keep it as a struct. I think in some of the early discussions/prototypes it was responsible for disposing and needed to be a class. Now, it's not disposable/mutable.